humanoid:
  # Joint State Controller
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 200

  # Ankle Controllers (Effort-based for balance control)
  ankle_pitch_left_effort_controller:
    type: effort_controllers/JointEffortController
    joint: ankle_pitch_left
    pid: {p: 50.0, i: 0.1, d: 5.0}

  ankle_pitch_right_effort_controller:
    type: effort_controllers/JointEffortController
    joint: ankle_pitch_right
    pid: {p: 50.0, i: 0.1, d: 5.0}

  ankle_roll_left_effort_controller:
    type: effort_controllers/JointEffortController
    joint: ankle_roll_left
    pid: {p: 30.0, i: 0.1, d: 5.0}

  ankle_roll_right_effort_controller:
    type: effort_controllers/JointEffortController
    joint: ankle_roll_right
    pid: {p: 30.0, i: 0.1, d: 5.0}

  # Hip Controllers (Effort-based for torque control)
  hip_pitch_left_effort_controller:
    type: effort_controllers/JointEffortController
    joint: hip_pitch_left
    pid: {p: 50.0, i: 0.1, d: 5.0}

  hip_pitch_right_effort_controller:
    type: effort_controllers/JointEffortController
    joint: hip_pitch_right
    pid: {p: 50.0, i: 0.1, d: 5.0}

  hip_roll_left_effort_controller:
    type: effort_controllers/JointEffortController
    joint: hip_roll_left
    pid: {p: 30.0, i: 0.1, d: 5.0}

  hip_roll_right_effort_controller:
    type: effort_controllers/JointEffortController
    joint: hip_roll_right
    pid: {p: 30.0, i: 0.1, d: 5.0}

  # Knee Controllers (Effort-based for torque control)
  knee_left_effort_controller:
    type: effort_controllers/JointEffortController
    joint: knee_left
    pid: {p: 60.0, i: 0.1, d: 10.0}

  knee_right_effort_controller:
    type: effort_controllers/JointEffortController
    joint: knee_right
    pid: {p: 60.0, i: 0.1, d: 10.0}

  gazebo_ros_control:
    pid_gains:
      hip_pitch_left:
        p: 50.0
        i: 0.1
        d: 5.0
      hip_pitch_right:
        p: 50.0
        i: 0.1
        d: 5.0
      hip_roll_left:
        p: 30.0
        i: 0.1
        d: 5.0
      hip_roll_right:
        p: 30.0
        i: 0.1
        d: 5.0
      knee_left:
        p: 60.0
        i: 0.1
        d: 10.0
      knee_right:
        p: 60.0
        i: 0.1
        d: 10.0
